src/open_inwoner/js/components/FilterBar/multiselect_listbox_checkbox.js

```diff
@@ -2,134 +2,225 @@
+      )
 
-  if (listboxDropdown) {
-    listboxDropdown.addEventListener('keydown', (e) => {
-      if (e.key === 'ArrowDown') {
-        e.preventDefault()
-        currentIndex = (currentIndex + 1) % listboxDropdown.children.length
-        listboxDropdown.children[currentIndex].focus()
-      } else if (e.key === 'ArrowUp') {
-        e.preventDefault()
-        currentIndex =
-          (currentIndex - 1 + listboxDropdown.children.length) %
-          listboxDropdown.children.length
-        listboxDropdown.children[currentIndex].focus()
-      } else if (e.key === 'Escape') {
-        listboxDropdown.classList.remove('show')
-        selectButton.setAttribute('aria-expanded', 'false')
-        selectButton.focus()
+      checkboxes.forEach((checkbox) => {
+        const value = checkbox.value
+        checkbox.checked = urlParams.getAll('status').includes(value)
+      })
+
+      calculateAndDisplayCheckedSum() // Update button and sum
+    }
+
+    const calculateAndDisplayCheckedSum = function () {
+      const checkboxes = document.querySelectorAll(
+        '.filter-bar .checkbox__input'
+      )
+      let sum = 0
+      let selectedFilters = []
+
+      /**
+       * Note: dynamic text from selectedFilters is inserted using textContent,
+       * while static elements like icons or spans are appended separately using createElement and appendChild.
+       * ensuring HTML injection is safe and prevents cross-site scripting vulnerability.
+       * */
+
+      checkboxes.forEach((checkbox) => {
+        if (checkbox.checked) {
+          const label = checkbox.nextElementSibling
+          selectedFilters.push(label.textContent.trim())
+          const frequencyCounter = label.querySelector('.frequency-counter')
+          const match = frequencyCounter.textContent.match(/\d+/)
+          const value = match ? parseInt(match[0]) : 0
+
+          if (!isNaN(value)) {
+            sum += value
+          }
+        }
+      })
+
+      const selectButton = document.getElementById('selectButton')
+      selectButton.innerHTML = '' // Clear the button content before appending new elements
+
+      let expandIcon = document.createElement('span')
+      expandIcon.classList.add('material-icons')
+      expandIcon.setAttribute('aria-hidden', 'true')
+      expandIcon.textContent = 'expand_more'
+
+      let closeIcon = document.createElement('span')
+      closeIcon.classList.add('material-icons', 'close-icon')
+      closeIcon.setAttribute('aria-hidden', 'true')
+      closeIcon.textContent = 'close'
+
+      if (selectedFilters.length === 0) {
+        // Show 'Status' text and 'expand_more' icon when no filters are selected
+        selectButton.textContent = 'Status '
+        selectButton.appendChild(expandIcon)
+        selectButton.classList.remove('active')
+      } else {
+        if (selectedFilters.length === 1) {
+          // Show only the selected filter text and close icon
+          selectButton.textContent = ''
```

Comment:
In the design the 'status' phrase for the situation when only 1 is selected, was left out intentionally - perhaps because some of them have a very long name. So I'm keeping this out of the displayed selection for now. This component has not been tested in real life and I feel like we could benefit from user feedback.